lib/ldp.js

```diff
@@ -256,25 +256,53 @@ class LDP {
     })
   }
 
-  put (host, resourcePath, stream, callback) {
+  isValidRdf (body, requestUri, contentType) {
+    const resourceGraph = $rdf.graph()
+    try {
+      $rdf.parse(body, resourceGraph, requestUri, contentType)
+    } catch (err) {
+      debug.ldp('VALIDATE -- Error parsing data: ' + err)
+      return false
+    }
+    return true
+  }
+
+  _put (host, resourcePath, callback, resourceHandler) {
     const ldp = this
     const root = !ldp.multiuser ? ldp.root : ldp.root + host + '/'
     const filePath = utils.uriToFilename(resourcePath, root, host)
 
     // PUT requests not supported on containers. Use POST instead
     if (filePath.endsWith('/')) {
-      return callback(error(409,
-        'PUT not supported on containers, use POST instead'))
+      return callback(error(409, 'PUT not supported on containers, use POST instead'))
     }
+
     // First, create the enclosing directory, if necessary
     const dirName = path.dirname(filePath)
     mkdirp(dirName, (err) => {
       if (err) {
         debug.handlers('PUT -- Error creating directory: ' + err)
-        return callback(error(err,
-          'Failed to create the path to the new resource'))
+        return callback(error(err, 'Failed to create the path to the new resource'))
       }
+
       // Directory created, now write the file
+      resourceHandler(filePath)
+    })
+  }
+
+  putText (host, resourcePath, text, callback) {
+    this._put(host, resourcePath, callback, (filePath) => {
+      try {
+        fs.writeFileSync(filePath, text)
+      } catch (err) {
+        return callback(error(500, err.message))
```

Comment:
You might want to delete whatever was written do disk.

Reply:
What if the file existed from before, and we need to recover that file?

If it exists from before, maybe I should do a backup of the content, and roll back if the write failed? Otherwise, I should delete whatever was written to disk?